sys/include/net/gnrc/lorawan.h

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2017 Fundación Inria Chile
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_lorawan GNRC LoRaWAN
+ * @ingroup     net_gnrc
+ * @brief       GNRC LoRaWAN stack implementation
+ *
+ * @{
+ *
+ * @file
+ * @brief   GNRC LoRaWAN API definition
+ *
+ * @author  José Ignacio Alamos <jose.alamos@haw-hamburg.de>
+ * @author  Francisco Molina <femolina@uc.cl>
+ */
+#ifndef NET_GNRC_LORAWAN_H
+#define NET_GNRC_LORAWAN_H
+
+#include "gnrc_lorawan_internal.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief maximum timer drift in percentage
+ *
+ * E.g a value of 0.1 means there's a positive drift of 0.1% (set timeout to
+ * 1000 ms => triggers after 1001 ms)
+ */
+#ifndef CONFIG_GNRC_LORAWAN_TIMER_DRIFT
+#define CONFIG_GNRC_LORAWAN_TIMER_DRIFT 1
+#endif
+
+/**
+ * @brief the minimum symbols to detect a LoRa preamble
+ */
+#ifndef CONFIG_GNRC_LORAWAN_MIN_SYMBOLS_TIMEOUT
+#define CONFIG_GNRC_LORAWAN_MIN_SYMBOLS_TIMEOUT 50
+#endif
+
+/**
+ * @brief enable or disable low power mode callbacks
+ */
+#ifndef CONFIG_GNRC_LORAWAN_ENABLE_LPM_STORAGE
+#define CONFIG_GNRC_LORAWAN_ENABLE_LPM_STORAGE (0)
+#endif
+
+
+#define GNRC_LORAWAN_REQ_STATUS_SUCCESS (0)     /**< MLME or MCPS request successful status */
+#define GNRC_LORAWAN_REQ_STATUS_DEFERRED (1)    /**< the MLME or MCPS confirm message is asynchronous */
+
+/**
+ * @brief MAC Information Base attributes
+ */
+typedef enum {
+    MIB_ACTIVATION_METHOD
+} mlme_mib_type_t;
+
+/**
+ * @brief MAC Information Base descriptor for MLME Request-Confirm
+ */
+typedef struct {
+    mlme_mib_type_t type; /**< MIB attribute identifier */
```

Comment:
where do the following struct definitions come from? Because they look a bit inconsistent in style, i.e., some use an `enum` for `type` other generic `uint8_t`. Also the order is different, i.e., `param` after `type` or before. If possible, this should be made consistent.

Reply:
I unified them as enums. See last commits